ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java

```diff
@@ -724,6 +746,17 @@ static void setRequestTimeout(RaftProperties properties, TimeDuration timeoutDur
       setTimeDuration(properties::setTimeDuration, REQUEST_TIMEOUT_KEY, timeoutDuration);
     }
 
+    String STREAM_REQUEST_TIMEOUT_KEY = PREFIX + ".stream.request.timeout";
```

Comment:
Let's move it to `GrpcConfigKeys`.  We have a `LEADER_OUTSTANDING_APPENDS_MAX_KEY` there.  Then, we can add `LEADER_OUTSTANDING_INSTALL_SNAPSHOT_REQUESTS_MAX_KEY` and uses it to compute the timeout inside the code.  BTW, we also need it for limiting the number of outstanding requests later.

Reply:
> ... BTW, we also need it for limiting the number of outstanding requests later.

Filed https://issues.apache.org/jira/browse/RATIS-1812 .

Reply:
Thanks for the review! Will make the changes after https://github.com/apache/ratis/pull/850 is merged.